src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -65,8 +71,22 @@ class Attribute {
     }
 
     // Apply callback to materialize RIC inside graph
-    void operator()(Input<Node> input) const {
-        m_callback(input, *this);
+    void materialize(Input<Node> input) const {
```

Comment:
do we use `m_callback` somewhere else?  can we also delete `set_callback` method and related things?

just in case, are you sure that we don't need this `m_callback` and this method is the same for all operations?